client/driver/logging/rotator_test.go

```diff
@@ -168,6 +169,72 @@ func TestFileRotator_RotateFiles(t *testing.T) {
 	})
 }
 
+func TestFileRotator_RotateFiles_Boundary(t *testing.T) {
+	t.Parallel()
+	var path string
+	var err error
+	if path, err = ioutil.TempDir("", pathPrefix); err != nil {
+		t.Fatalf("test setup err: %v", err)
+	}
+	defer os.RemoveAll(path)
+
+	fr, err := NewFileRotator(path, baseFileName, 10, 5, logger)
+	if err != nil {
+		t.Fatalf("test setup err: %v", err)
+	}
+
+	// We will write two times:
+	// 1st: Write with new lines spanning two files
+	// 2nd: Write long string with no new lines
+	expectations := [][]byte{
+		[]byte("ab\n"),
+		[]byte("cdef\n"),
```

Comment:
would suggest adding test cases with new line in the beginning, and middle of string

Reply:
str1 captures the middle but I can add one for the beginning